src/v/cluster/topic_table.cc

```diff
@@ -383,6 +384,60 @@ topic_table::apply(cancel_moving_partition_replicas_cmd cmd, model::offset o) {
     co_return errc::success;
 }
 
+ss::future<std::error_code>
+topic_table::apply(move_topic_replicas_cmd cmd, model::offset o) {
+    auto tp = _topics.find(model::topic_namespace_view(cmd.key));
+    if (tp == _topics.end()) {
+        co_return errc::topic_not_exists;
+    }
+    if (!tp->second.is_topic_replicable()) {
+        co_return errc::topic_operation_error;
+    }
+
+    // We should check partition before create updates
+
+    if (std::any_of(
+          cmd.value.begin(),
+          cmd.value.end(),
+          [&tp](const auto& partition_and_replicas) {
+              return !tp->second.get_assignments().contains(
+                partition_and_replicas.partition);
+          })) {
+        vlog(
+          clusterlog.error,
+          "Can not move replicas for topic({}), becasue can not find "
+          "partitions",
+          cmd.key);
+        co_return errc::partition_not_exists;
+    }
+
+    if (std::any_of(
+          cmd.value.begin(),
+          cmd.value.end(),
+          [this, key = cmd.key](const auto& partition_and_replicas) {
+              return _updates_in_progress.contains(
+                model::ntp(key.ns, key.tp, partition_and_replicas.partition));
+          })) {
+        vlog(
+          clusterlog.error,
+          "Can not move replicas for topic({}), some updates in progress",
```

Comment:
Is this is a non-fatal thing that happens when the user submits a few overlapping operations at the same time?  If so, it should probably be logged at WARN instead of ERROR.